modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java

```diff
@@ -650,6 +660,49 @@ private void _addExtensionFilterLabelItems(
 		}
 	}
 
+	private Set<String> _getAssetTagIds(HttpServletRequest httpServletRequest) {
+		if (_assetTagIds != null) {
+			return _assetTagIds;
+		}
+
+		_assetTagIds = new HashSet(
+			Arrays.asList(
+				ArrayUtil.toStringArray(
+					ParamUtil.getStringValues(
+						httpServletRequest, "assetTagId"))));
+
+		return _assetTagIds;
+	}
```

Comment:
Shouldn't this be named `getSelectedTagNames` based on its usage?

Also, why are you calling `ArrayUtil.toStringArray`? `getStringValues` already returns a string array.

Reply:
Even simpler, directly use `SetUtil.fromArray()`:

```java
		_assetTagIds = SetUtil.fromArray(
			ParamUtil.getStringValues(
				httpServletRequest, "assetTagId"));
```